src/Neo/Network/P2P/Payloads/Conditions/GroupCondition.cs

```diff
@@ -66,5 +90,21 @@ public override StackItem ToStackItem(IReferenceCounter referenceCounter)
             result.Add(Group.ToArray());
             return result;
         }
+
+        public static bool operator ==(GroupCondition left, GroupCondition right)
+        {
+            if (((object)left) == null || ((object)right) == null)
+                return Equals(left, right);
+
+            return left.Equals(right);
+        }
+
+        public static bool operator !=(GroupCondition left, GroupCondition right)
+        {
+            if (((object)left) == null || ((object)right) == null)
+                return !Equals(left, right);
+
+            return !(left.Equals(right));
```

Comment:
Remove all extra parenthesis and use `[MethodImpl(MethodImplOptions.AggressiveInlining)]` in the operators

Reply:
we need the parenthesis to check for `null` so it knows what `Equals` to use.